x/nexus/keeper/general_message.go

```diff
@@ -73,8 +107,11 @@ func (k Keeper) SetMessageExecuted(ctx sdk.Context, messageID exported.MessageID
 		return fmt.Errorf("general message %s not found", messageID.String())
 	}
 
-	if !m.Is(exported.Sent) {
-		return fmt.Errorf("general message is not sent")
+	if !(m.Is(exported.Sent) || m.Is(exported.Approved)) {
```

Comment:
i prefer changing the status we set in call contract to be `Sent` instead of `Approved`. This should keep the checks here cleaner. Naming isn't ideal either way, since both Sent/Approved is used before the actual approval is created in evm lol

Reply:
@haiyizxx when exactly are messages in the `Sent` state?

Reply:
I'm ok changing it to `Sent`, though that means `GetApprovedMessages` will need to change to `GetSentMessages` and the secondary table will store sent message IDs instead of approved message IDs. This will affect the flow of any messages being consumed in the end blocker, regardless of where they originate.

Reply:
I changed the status set by call contract to `Sent`. I do think this makes the state transitions a little clearer